File: sentimental-credit/credit.py
cardnum = input("Credit card number: ")
while not cardnum.isdigit():
    cardnum = input("Credit card number: ")

counter = 1
total = 0
length = len(cardnum)
cardnumres = int(cardnum)
cardnum_copy = int(cardnum)


while cardnum_copy > 0:
    remainder = cardnum_copy % 10
    cardnum_copy = cardnum_copy // 10

    if counter % 2 == 0:
        remainder *= 2
        if remainder > 9:
            remainder = (remainder % 10) + (remainder // 10)

    total += remainder
    counter += 1

if total % 10 == 0:
    start_digits = int(str(cardnumres)[:2])
    first_digit = int(str(cardnumres)[0])

    if (start_digits == 34 or start_digits == 37) and length == 15:
        print("AMEX")
    elif 51 <= start_digits <= 55 and length == 16:
        print("MASTERCARD")
    elif first_digit == 4 and length in [13, 16]:
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main (void)
{
    long x = get_long("credit card number: ");

    int sum = 0;
    int c = 1;
    long long z = x;
    int cl = 0;

    while (x != 0)
    {
        int r = x % 10;
        x = x / 10;

        if (c % 2 == 0)
        {
            r = r * 2;
            if (r > 9)
            {
                r = (r % 10) + (r / 10);
            }
        }
        sum = sum + r;
        c++;
        cl++;
    }
    if (sum % 10 == 0)
    {
        while (z >= 100)
        {
            z /= 10;
        }
        if ((z == 34 || z == 37) && (cl == 15))
        {
            printf("AMEX\n");
        }
        else if ((z >= 51 && z <= 55) && (cl == 16))
        {
            printf("MASTERCARD\n");
        }
        else if ((z / 10 == 4 || z == 4) && (cl == 13 || cl == 16))
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}
